app/k9mail/src/main/java/com/fsck/k9/widget/unread/UnreadWidgetConfigurationFragment.kt

```diff
@@ -46,18 +46,23 @@ class UnreadWidgetConfigurationFragment : PreferenceFragmentCompat() {
         }
 
         unreadFolderEnabled = findPreference(PREFERENCE_UNREAD_FOLDER_ENABLED)!!
-        unreadFolderEnabled.onPreferenceChangeListener = Preference.OnPreferenceChangeListener { _, _ ->
-            unreadFolder.summary = getString(R.string.unread_widget_folder_summary)
-            selectedFolderId = null
-            true
+        if (selectedAccountUuid == null) {
+            unreadFolderEnabled.isEnabled = false
+            unreadFolderEnabled.isChecked = false
+        } else {
```

Comment:
This case is never executed. But it should. The listener is responsible for resetting the "Folder" preference description to the default when the "Folder count" preference is unchecked (and a folder was selected before).

Steps to test is works as intended (like it did before your change):
1. Select an account
2. Check "Folder count"
3. Tap "Folder" preference and select a folder -> Folder name is displayed as preference "description"
4. Uncheck "Folder count" -> Folder name "description" should be reset to default and no longer show the folder name